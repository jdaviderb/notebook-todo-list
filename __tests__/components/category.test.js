import React, { Component } from 'react';
import { shallow } from 'enzyme';
import Category from '../../app/components/category';

describe('Component: Category', () => {

  it('renders correctly', () => {
    const wrapper = shallow(<Category />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render with correctyle label', () => {
    const wrapper = shallow(<Category />);
    const label = 'My Category';
    wrapper.setProps({ label });
    const wrapperLabel = wrapper.find('Text').children().text();
    expect(wrapperLabel).toEqual(label);
  });

});