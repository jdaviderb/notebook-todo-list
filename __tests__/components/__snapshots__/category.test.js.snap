// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Category renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Category />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontSize": 25,
              },
              undefined,
            ]
          }
        />,
        <Component
          style={
            Object {
              "borderColor": "black",
              "borderWidth": 0.4,
              "flexDirection": "row",
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": undefined,
          "ellipsizeMode": "tail",
          "style": Array [
            Object {
              "fontSize": 25,
            },
            undefined,
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "borderColor": "black",
            "borderWidth": 0.4,
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 25,
                },
                undefined,
              ]
            }
          />,
          <Component
            style={
              Object {
                "borderColor": "black",
                "borderWidth": 0.4,
                "flexDirection": "row",
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": undefined,
            "ellipsizeMode": "tail",
            "style": Array [
              Object {
                "fontSize": 25,
              },
              undefined,
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "borderColor": "black",
              "borderWidth": 0.4,
              "flexDirection": "row",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
